### PR TITLE
Enable stitching in tests

### PR DESCRIPTION
**Pull request by orta**
The environment for running tests now accurately matches how we run stitching on staging, with stitching turned on for gravity and metaphysics as a local schema.

Right now this includes all of #1307 so is a bit messy -  only one commit matters. 
https://github.com/artsy/metaphysics/commit/1d6d914f73aca2d53a1ee278cf94cf7ec807c04d 

I'm gonna comment inline *on that commit*

### CODE REVIEW

File: src/lib/stitching/gravity/link.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const createGravityLink = () => {
     const locals = context.graphqlContext && context.graphqlContext.res.locals
     const headers = { ...(locals && requestIDHeaders(locals.requestIDs)) }
     Object.assign(headers, { "X-XAPP-TOKEN": config.GRAVITY_XAPP_TOKEN })
-    if (locals.accessToken) {
+    if (locals && locals.accessToken) {
```
[the remark this answers is not in the document]
Comment (alloy) [reply src/lib/stitching/gravity/link.ts]: Can’t we define `locals` from the `runQuery` test helper and more accurately mimic runtime code paths?

File: src/schema/__tests__/partner_show.test.js [hunk 2]
```diff
@@ -106,7 +106,7 @@ describe("PartnerShow type", () => {
         throw new Error("Did not expect query to not throw an error")
       })
       .catch(error => {
-        expect(error.statusCode).toEqual(404)
+        expect(error.message).toEqual("Show Not Found")
```
[the remark this answers is not in the document]
Comment (alloy) [reply src/schema/__tests__/partner_show.test.js]: I think it should be a GraphQL error with extra metadata that contains the status code.

[the remark this answers is not in the document]
Comment (alloy) [reply src/schema/__tests__/partner_show.test.js]: See https://github.com/artsy/lewitt-api/pull/308#issuecomment-380094636

File: src/schema/show.js [hunk 3]
```diff
@@ -422,7 +423,7 @@ const Show = {
     return showLoader(id)
       .then(show => {
         if (!show.displayable && !show.is_reference) {
-          return new HTTPError("Show Not Found", 404)
+          return new GraphQLError("Show Not Found")
```
[the remark this answers is not in the document]
Comment (alloy) [reply src/schema/show.js]: Honestly, this shouldn’t even be an error imo

[the remark this answers is not in the document]
Comment (alloy) [reply src/schema/show.js]: …but that’s not for this PR to change

File: src/lib/stitching/gravity/link.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (orta) [reply src/lib/stitching/gravity/link.ts]: Yeah, setting the default object as having a default version of that is a better call - will switch

File: _schema.graphql [hunk 4]
```diff
@@ -1958,7 +1958,7 @@ type Collection implements Node {
     before: String
     last: Int
     private: Boolean = false
-    sort: CollectionSorts = POSITION_DESC
+    sort: CollectionSorts
```
Comment (orta) [inline _schema.graphql]: These were conscious changes ^ 

Stitching didn't reliably show the right output here, so I didn't want to be relying on the implementation being hooked up correct. Instead. I moved the default setting to be at runtime.

File: _schema.graphql [hunk 5]
```diff
@@ -2575,6 +2575,23 @@ type dimensions {
...
+  given_name: String
+
+  # Uniq ID for this partner
+  id: ID!
+  relative_size: Int
+
+  # Uniq, friendly identifier for this artist
+  slug: String!
+  subscription_state: String
+}
```
Comment (orta) [inline _schema.graphql]: We had to expose something, so I took one class and renamed it

**Comment by orta** [comment]
OK, this is now ready to go 👍

File: _schema.graphql (hunk 4 above)
Comment (alloy) [reply _schema.graphql]: Did these defaults already have associated tests?

**Pull request merged**